embedchain/memory/base.py

```diff
@@ -73,21 +73,33 @@ def delete(self, app_id: str, session_id: Optional[str] = None):
         self.cursor.execute(DELETE_CHAT_HISTORY_QUERY, params)
         self.connection.commit()
 
-    def get(self, app_id, session_id, num_rounds=10, display_format=False) -> list[ChatMessage]:
+    def get(
+        self, app_id, session_id: str = "default", num_rounds=10, fetch_all: bool = False, display_format=False
+    ) -> list[ChatMessage]:
         """
         Get the most recent num_rounds rounds of conversations
         between human and AI, for a given app_id.
         """
 
-        QUERY = """
-            SELECT * FROM ec_chat_history
-            WHERE app_id=? AND session_id=?
-            ORDER BY created_at DESC
-            LIMIT ?
-        """
+        if fetch_all:
```

Comment:
You can refactor this code to avoid repetition like this:

```python
base_query = """
    SELECT * FROM ec_chat_history
    WHERE app_id=?
"""

if fetch_all:
    additional_query = "ORDER BY created_at DESC"
    params = (app_id,)
else:
    additional_query = "AND session_id=? ORDER BY created_at DESC LIMIT ?"
    params = (app_id, session_id, num_rounds)

QUERY = base_query + additional_query
```